### PR TITLE
Support RSV on Results page

### LINKED ISSUE #6987: [Single-Entry RSV] Results page supports RSV devices
## Why should we do this?

- Support for RSV must be added to the frontend single-entry user-facing experience of SimpleReport, to include RSV as part of Multiplex devices that test for RSV, as well as when an RSV-only device has been selected.

## Action Requested

- Update components on Results page to include support for RSV devices

## Acceptance Criteria
- The results page Condition filter includes RSV
- The "Download results" spreadsheet includes a column for RSV
- RSV is shown on the printed page for "Print details"
- RSV is shown on the modal for "View details"
- RSV is supported for emailing and texting results

### RSV missing from filter
Image

### RSV missing from View details
Image

### RSV missing from Print details
Image

### RSV missing from Download results spreadsheet
Image

### PR DESCRIPTION
**Pull request by mehansen**
# FRONTEND PULL REQUEST

## Related Issue

resolves #6987 

## Changes Proposed

* add RSV to the result page disease filter
* add an RSV column in our CSV for downloading results
* dynamically populate the "View details" modal with only the results that exist for the test event
* add RSV copy so it shows up in both the print result and pxp (email/text results)

## Additional Information

- copy for the header for print and pxp and the guidance (more info section) will be updated in #7000 

## Testing

will deploy to `dev3`

### CODE REVIEW

File: frontend/src/app/testResults/types.d.ts [hunk 1]
```diff
@@ -20,9 +20,6 @@ interface MultiplexResult extends DiseaseName {
 
 type MultiplexResults = MultiplexResult[];
 
-// Represents a single test result with context from parent test event
-type EnrichedMultiplexResult = TestResult & MultiplexResult;
```
Comment (mehansen) [inline frontend/src/app/testResults/types.d.ts:24]: this wasn't being used anywhere

File: frontend/src/app/utils/testResultCSV.ts [hunk 2]
```diff
@@ -79,5 +81,19 @@ export function parseDataForCSV(data: any[]) {
...
+          "RSV result":
+            TEST_RESULT_DESCRIPTIONS[
+              getResultByDiseaseName(r.results, "RSV") as Results
+            ],
+          ...csvData2,
+        }
+      : {
+          ...csvData1,
+          ...csvData2,
+        };
```
Comment (mehansen) [inline frontend/src/app/utils/testResultCSV.ts:97]: I wanted the RSV result column to come after the Covid, flu A, and flu B columns

called out clean up for this method in #7056

File: frontend/src/patientApp/PxpApiService.ts [hunk 3]
```diff
@@ -44,7 +44,6 @@ export type SelfRegistrationData = Omit<
 
 export type VerifyV2Response = {
   testEventId: string;
-  result: TestResult;
```
Comment (mehansen) [inline frontend/src/patientApp/PxpApiService.ts:47]: this wasn't being used or populated, as far as I can tell

File: frontend/src/app/testResults/viewResults/actionMenuModals/TestResultDetailsModal.test.tsx [hunk 4]
```diff
@@ -87,11 +91,10 @@ describe("non-multiplex TestResultDetailsModal", () => {
   });
 });
 
-describe("Multiplex TestResultDetailsModal", () => {
+describe("multiple diseases TestResultDetailsModal", () => {
   let component: any;
 
   beforeEach(() => {
-    jest.spyOn(flaggedMock, "useFeature").mockReturnValue(true);
```
Comment (mehansen) [inline frontend/src/app/testResults/viewResults/actionMenuModals/TestResultDetailsModal.test.tsx:94]: I think this was added to mock the multiplex feature flag

**Comment by mehansen** [comment]
oops I think the sorting on the view details modal is indeterminate 👀 
Screenshot 2023-12-07 at 11 57 55 AM

**Review by fzhao99 — APPROVED** [review]
lgtm! left one comment for a followup PR

File: frontend/src/app/testResults/viewResults/TestResultsList.test.tsx [hunk 5]
```diff
@@ -864,4 +865,36 @@ describe("TestResultsList", () => {
       );
     });
   });
+
+  describe("RSV single-entry flag is on", () => {
+    it("should show RSV option in disease drop-down", async () => {
+      const flagSpy = jest.spyOn(flaggedMock, "useFeature");
+      flagSpy.mockImplementation((flagName) => {
+        return flagName === "singleEntryRsvEnabled";
```
Comment (mpbrown) [inline frontend/src/app/testResults/viewResults/TestResultsList.test.tsx:873]: Very clean conditional mock!

**Review by mpbrown — APPROVED** [review]
LGTM!

> do we want to do a general audit/change of the site (including the static site) to make it more clear/accurate that it's not just COVID-19 tests we are handling?

^ this is a good point and I agree we should look into clarifying this

**Comment by mehansen** [comment]
>do we want to do a general audit/change of the site (including the static site) to make it more clear/accurate that it's not just COVID-19 tests we are handling?

Thanks @emyl3! Created #7084

File: frontend/src/app/utils/testResultCSV.ts [hunk 6]
```diff
@@ -11,11 +11,11 @@ import { getResultByDiseaseName } from "./testResults";
 
 import { displayFullName, facilityDisplayName } from "./index";
 
-export function parseDataForCSV(data: any[]) {
+export function parseDataForCSV(includeRSV: any, data: any[]) {
```
[the remark this answers is not in the document]
Comment (mehansen) [reply frontend/src/app/utils/testResultCSV.ts:14]: yes agree this data needs to be typed! adding that to the note for cleanup in #7056

**Pull request merged**